amethyst_animation/src/material.rs

```diff
@@ -170,27 +170,29 @@ impl AnimationSampling for Material {
         channel: &Self::Channel,
         extra: &Read<MaterialTextureSet>,
     ) -> Self::Primitive {
+        const ERR_MSG: &'static str = "Unable to get requested channel from MaterialTextureSet.";
+
         match *channel {
             MaterialChannel::AlbedoTexture => {
-                MaterialPrimitive::Texture(extra.id(&self.albedo).unwrap())
+                MaterialPrimitive::Texture(extra.id(&self.albedo).expect(ERR_MSG))
```

Comment:
Would require altering the trait interface.  Not sure if that's desirable or not.

Reply:
So this would only be triggered in the following scenario:

* You start at texture animation for a specific texture, say albedo
* The animation have a sampler for the albedo for a specific entity
* Said entity does not already have a albedo texture defined in its material

If we want to change the return value of this function, I think it should be changed to an Option, not Result.